src/coreclr/src/jit/error.h

```diff
@@ -153,24 +154,17 @@ extern void RecordNowayAssertGlobal(const char* filename, unsigned line, const c
 #define unreached() noWayAssertBody()
 
 #define NOWAY_MSG(msg) noWayAssertBodyConditional(NOWAY_ASSERT_BODY_ARGUMENTS)
+#define NOWAY_MSG_FILE_AND_LINE(msg, file, line) noWayAssertBodyConditional(NOWAY_ASSERT_BODY_ARGUMENTS)
 
 #endif // !DEBUG
 
 
 #if 1 // All platforms currently enable NYI; this should be a tighter condition to exclude some platforms from NYI
 
-#if defined(ALT_JIT)
-
 // This can return based on Config flag/Debugger
 extern void notYetImplemented(const char* msg, const char* file, unsigned line);
 #define NYIRAW(msg) notYetImplemented(msg, __FILE__, __LINE__)
 
-#else // !defined(ALT_JIT)
-
-#define NYIRAW(msg) NOWAY_MSG(msg)
-
-#endif // !defined(ALT_JIT)
-
 #define NYI(msg)                    NYIRAW("NYI: " msg)
```

Comment:
We really should define NYI as a compile-time error for "complete" platforms (which now means all of them), but that's a separate issue.